silkrpc/commands/erigon_api.cpp

```diff
@@ -361,4 +359,42 @@ boost::asio::awaitable<void> ErigonRpcApi::handle_erigon_block_number(const nloh
     co_return;
 }
 
+// https://eth.wiki/json-rpc/API#erigon_cumulativeChainTraffic
+boost::asio::awaitable<void> ErigonRpcApi::handle_erigon_cumulative_chain_traffic(const nlohmann::json& request, nlohmann::json& reply) {
+    const auto params = request["params"];
+    if (params.size() != 1) {
+        auto error_msg = "invalid erigon_cumulativeChainTraffic params: " + params.dump();
+        SILKRPC_ERROR << error_msg << "\n";
+        reply = make_json_error(request["id"], 100, error_msg);
+        co_return;
+    }
+    const auto block_id = params[0].get<std::string>();
+
+    SILKRPC_DEBUG << "block_id: " << block_id << "\n";
+
+    auto tx = co_await database_->begin();
+
+    ChainTraffic chain_traffic;
+
+    try {
+        ethdb::TransactionDatabase tx_database{*tx};
+
+        const auto block_number = co_await core::get_block_number(block_id, tx_database);
+        chain_traffic.cumulative_transactions_count = co_await core::rawdb::read_cumulative_transaction_count(tx_database, block_number);
+        chain_traffic.cumulative_gas_used  = co_await core::rawdb::read_cumulative_gas_used(tx_database, block_number);
+
+        reply = make_json_content(request["id"], chain_traffic);
+    } catch (const std::exception& e) {
+        SILKRPC_ERROR << "exception: " << e.what() << " processing request: " << request.dump() << "\n";
+        reply = make_json_content(request["id"], chain_traffic);
+    } catch (...) {
+        SILKRPC_ERROR << "unexpected exception processing request: " << request.dump() << "\n";
+        reply = make_json_error(request["id"], 100, "unexpected exception");
+    }
```

Comment:
The JSON content returned in case of exception is kind of strange because it is a valid JSON result with an empty chain traffic data in case of any `std::exception`, whilst it is a JSON error in case of any other unorthodox exception.

Is this by any chance required to match Erigon behavior? If not, I would suggest a more consistent error handling.

Reply:
Yes to match erigon behaviour. In case of empty block the  response contains valid numbers not error.